src/v2/BaseLoginRouter.js

```diff
@@ -215,19 +210,29 @@ export default Router.extend({
         .then(idxResp => {
           this.settings.unset('stateToken');
           this.settings.unset('proxyIdxResponse');
-          this.settings.unset('useInteractionCodeFlow');
           this.appState.trigger('remediationSuccess', idxResp);
-          this.render(Controller, options);
         })
         .catch(errorResp => {
           this.settings.unset('stateToken');
           this.settings.unset('proxyIdxResponse');
-          this.settings.unset('useInteractionCodeFlow');
           this.appState.trigger('remediationError', errorResp.error || errorResp);
-          this.render(Controller, options);
+        })
+        .finally(() => {
+          this._render(Controller, options);
         });
     }
 
+    this._render(Controller, options);
```

Comment:
if we are doing the `render` in finally why we need it here as well?

Reply:
The problem I'm trying to solve is the recursive call of `this.render` in then/catch.
If we not unset `useInteractionCodeFlow `, `startLoginFlow` will be called again and result in infinite loop.
`useInteractionCodeFlow ` setting was loop-breaker.

Instead of this, I moved actual render logic into `_render` method and call it in `finally`.

Reply:
there seems to be some flaw with this flow, i am not super confident with this PR.. @aarongranick-okta you should take a look on why `useInteractionCodeFlow` unset is a loop breaker. It should not have that dependency.

Reply:
It seems that `render` is following a recursive pattern, used here to load languages:
https://github.com/okta/okta-signin-widget/blob/master/src/v2/BaseLoginRouter.js#L206

The lines of code to "unset" stateToken, etc. seem to be related to this function, which is not designed to be called recursively.
https://github.com/okta/okta-signin-widget/blob/master/src/v2/client/startLoginFlow.js#L26

Conceptually we want render to move forward, not in a loop. One way to accomplish this would be to use `async/await`

- await LanguageUtil.loadLanguage()
- await startLoginFlow
- continue with normal render

Although some methods return Q promises, I believe it should work. (Some tests may need minor adjustments).